commercetools-models/src/test/java/io/sphere/sdk/shippingmethods/queries/ShippingMethodsByCartGetIntegrationTest.java

```diff
@@ -2,20 +2,34 @@
 
 import io.sphere.sdk.carts.Cart;
 import io.sphere.sdk.shippingmethods.ShippingMethod;
+import io.sphere.sdk.shippingmethods.ShippingRate;
+import io.sphere.sdk.shippingmethods.ZoneRate;
 import io.sphere.sdk.test.IntegrationTest;
+import org.assertj.core.api.Condition;
 import org.junit.Test;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 import static io.sphere.sdk.carts.CartFixtures.*;
+import static org.assertj.core.api.Assertions.assertThat;
 
 
 public class ShippingMethodsByCartGetIntegrationTest extends IntegrationTest {
     @Test
     public void execution() throws Exception {
         final Cart cart = createCartWithShippingAddress(client());
 
-        final List<ShippingMethod> result =
+        final List<ShippingMethod> shippingMethods =
                 client().executeBlocking(ShippingMethodsByCartGet.of(cart));
+        assertThat(shippingMethods).isNotEmpty();
+
+        for (final ShippingMethod shippingMethod : shippingMethods) {
+            final List<ShippingRate> shippingRates = shippingMethod.getZoneRates().stream()
+                    .flatMap(zoneRate -> zoneRate.getShippingRates().stream())
+                    .collect(Collectors.toList());
+
+            assertThat(shippingRates).areExactly(1, new Condition<>(ShippingRate::isMatching, "Shipping rate is matching"));
```

Comment:
Maybe the assertion can be cooler with an `extracting`? It's a bit tricky because of the double list though, I haven't tried and probably it's not possible though.